docs/data/data-grid/list-view/components/ListCell.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { GridDensity, GridRenderCellParams } from '@mui/x-data-grid-premium';
+import Box from '@mui/material/Box';
+import { Card, CardContent, CardDetailList, CardDetail, CardTitle, CardMedia } from './Card';
+import { FileIcon } from './FileIcon';
+import { formatDate, formatSize } from '../utils';
+import { RowModel } from '../types';
+
+const ICON_SIZE_BY_DENSITY: Record<GridDensity, number> = {
+  compact: 24,
+  standard: 32,
+  comfortable: 16,
+};
+
+function Thumbnail(props: { fileIcon: React.ReactNode }) {
+  const { fileIcon } = props;
+  return (
+    <Box
+      sx={{
+        position: 'relative',
+        backgroundColor: 'grey.200',
+        borderRadius: 1,
+        width: 64,
+        height: 64,
+        overflow: 'hidden',
+      }}
+    >
+      <Box
+        sx={{
+          position: 'absolute',
+          bottom: 4,
+          left: 4,
+          display: 'flex',
+        }}
+      >
+        {fileIcon}
+      </Box>
+    </Box>
+  );
+}
+
+export function ListCell(params: GridRenderCellParams<RowModel>) {
+  const {
+    density,
+    columns: { columnVisibilityModel },
+  } = params.api.state;
```

Comment:
```suggestion
  const apiRef = useGridApiContext();
  const columnVisibilityModel = useGridSelector(
    apiRef,
    gridColumnVisibilityModelSelector,
  );
  const density = useGridSelector(apiRef, gridDensitySelector);
```

We want ListCell to rerender when the column visibility model changes, hence the usage of `useGridSelector`.
Fixes this issue:


https://github.com/user-attachments/assets/f5cfa89a-ee85-4c0f-a264-90e614945c21